### PR TITLE
Restaurant Manager and Owner Schema, Model, and Methods

### PR DESCRIPTION
**Pull request by goyaladit**
Added functionality to retrieve and post restaurantOwners and restaurantManagers

### CODE REVIEW

File: App.ts [hunk 1]
```diff
@@ -44,62 +49,56 @@ class App {
...
+    router.get('/app/restaurantmanagers/', (req, res) => {
+      console.log('Query All Restaurant Managers');
+      this.RestaurantManagers.retrieveAllRestaurantManagers(res);
+    });
+    router.get('/app/restaurantowners/', (req, res) => {
+      console.log('Query All Restaurant Owners');
+      this.RestaurantOwners.retrieveAllRestaurantOwners(res);
+    });
+    router.post('/app/restaurantmanagers/create-manager', (req, res) => {
+      //TODO: add prechecks
```
Comment (Nirmalya24) [inline App.ts:66]: Precheck is missing. It should check the restaurantOwnerID passed is valid

File: App.ts [hunk 2]
```diff
@@ -44,62 +49,56 @@ class App {
...
+
+    router.post('/app/restaurantmanagers/create-owner', (req, res) => {
+      //TODO: add prechecks
+
+      //params from request body
+      let createOwner: object = {
+        userID: crypto.randomUUID(),
+        password: req.body.password,
+        connectStatus: req.body.connectStatus,
+        //TODO: might have to convert req.body.restaurantList from JSON to RestaurantModel type list
```
Comment (Nirmalya24) [inline App.ts:93]: Convert the Schema/Model to Object type

File: model/RestaurantManagerModel.ts [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+            {  
+                //IUsersModel
+                userID:String,
+                password:String,
+                connectStatus:Boolean,
+                //IRestaurantManagerModel
+                managerName: String,
+                restaurantOwnerID:String,
+                restuarantID:String,
+            }, {collection: 'Restaurant Managers'}
```
Comment (Nirmalya24) [inline model/RestaurantManagerModel.ts:29]: Remove the space from collection name. Rename it to RestaurantManagers

Comment (Nirmalya24) [reply model/RestaurantManagerModel.ts:29]: Rename subsequent occurrences as well.

File: model/RestaurantOwnerModel.ts [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    public createSchema():void {
+        this.schema = new Mongoose.Schema (
+            {  
+                //IUsersModel
+                userID:String,
+                password:String,
+                connectStatus:Boolean,
+                //IRestaurantOwnerModel
+                restaurantList: [{type: Mongoose.Schema.Types.ObjectId, ref: 'RestaurantModel'}]
+            }, {collection: 'Restaurant Owners'}
```
Comment (Nirmalya24) [inline model/RestaurantOwnerModel.ts:27]: Remove space from collection name and update subsequent occurrences.

**Review by Nirmalya24 — CHANGES_REQUESTED** [review]

File: App.ts (hunk 1 above)
Comment (goyaladit) [reply App.ts:66]: Can you explain what you mean by valid? (are you talking about the case where the restaurantOwnerID passed is already present?)

Comment (goyaladit) [reply App.ts:66]: note: we decided to only that the restaurantOwnerID field is not empty

File: App.ts (hunk 2 above)
Comment (goyaladit) [reply App.ts:93]: note: we converted restaurantList to type string array

**Review by Nirmalya24 — APPROVED** [review]
LGTM!

**Pull request merged**